skema/skema-rs/mathml/src/expression.rs

```diff
@@ -14,42 +25,1172 @@ enum Atom {
 #[derive(Debug, PartialEq, Clone)]
 enum Expr {
     Atom(Atom),
-    Expression { op: Operator, args: Vec<Expr> },
+    Expression {
+        op: Vec<Operator>,
+        args: Vec<Expr>,
+        name: String,
+    },
+}
+
+/// Intermediate data structure to support the generation of graphs of mathematical expressions
+#[derive(Debug, PartialEq, Clone)]
+pub struct PreExp {
+    op: Vec<Operator>,
+    args: Vec<Expr>,
+    name: String,
 }
 
 impl MathExpression {
-    fn to_expr(&self) -> Expr {
+    pub fn to_expr(self, mut pre: &mut PreExp) {
         match self {
-            Mi(x) => Expr::Atom(Atom::Identifier(x.clone())),
-            Mo(x) => Expr::Atom(Atom::Operator(x.clone())),
-            Mrow(xs) => Expr::Expression {
-                op: Operator::Add,
-                args: Vec::<Expr>::new(),
-            },
+            Mi(x) => {
+                if pre.args.len() >= pre.op.len() {
+                    /// deal with the invisible multiply operator
+                    pre.op.push(Operator::Multiply);
+                }
+                pre.args.push(Expr::Atom(Atom::Identifier(x.to_string())));
+            }
+            Mn(x) => {
+                if pre.args.len() >= pre.op.len() {
+                    /// deal with the invisible multiply operator
```

Comment:
Doc comments (starting with triple slashes `///`) are not meant to be in the interior of a function - they are the Rust equivalent of Python docstrings, and should only be there at the top of a struct/function/method/module.

For regular comments inside a function, you can use a double slash (`//`)

Reply:
I see. Thanks for letting me know! I will replace them.